src/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs

```diff
@@ -135,15 +106,14 @@ public override bool Equals(object obj)
                 return true;
             }
 
-            DesignerAttribute other = obj as DesignerAttribute;
+            if (!(obj is DesignerAttribute other))
+            {
+                return false;
```

Comment:
Why break it apart like this, rather than just doing what the original was doing:
```C#
return obj is DesignerAttribute other && other.DesignerBaseTypeName == DesignerBaseTypeName && other.DesignerTypeName == DesignerTypeName;
```
?  FWIW, I find the extra `if` version harder to read.

Reply:
Fixed, you're right